rlp/src/main/scala/io/iohk/ethereum/rlp/package.scala

```diff
@@ -44,6 +44,23 @@ package object rlp {
     override def toString: String = s"RLPValue(${Hex.toHexString(bytes)})"
   }
 
+  /** Modelise a RLPEncodable that should be binary prefixed by a raw byte.
+    *
+    * When converting this RLPEncodable to byte, the resulting value will be:
+    * prefix || prefixedRLPEncodable.toByte
+    * where || is the binary concatenation symbol.
+    *
+    * To be able to read back the data, use TypedTransaction.TypedTransactionsRLPAggregator
+    *
+    * This is for example used for typed transaction and typed receipt.
+    *
+    * @param prefix the raw byte
+    * @param prefixedRLPEncodeable the RLPEncodable to prefix with
+    */
+  case class PrefixedRLPEncodable(prefix: Byte, prefixedRLPEncodeable: RLPEncodeable) extends RLPEncodeable {
+    require(prefix <= 0x07f, "prefix should be lower than 0x7f")
```

Comment:
```
scala> 0x07f.toByte
val res0: Byte = 127

scala> 128.toByte <= 0x7f.toByte
val res3: Boolean = true
```

Looks like all `Byte`s are smaller than `0x7f` so it makes more sense to check whether `prefix` is larger than 0

Reply:
good catch, I'll change the check and add proper testing